# Creates summary p-value files for use with MAGMA

# Subset file to the following white space seperated format: RSID CHR BP P N

# EGG cohorts. Needed to remove some erroneous rows in the birth length summary statitsics file.
awk '{print $3,$1,$2,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_birth_length/EGG-GWAS-BL.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_birth_length/EGG-GWAS-BL_MAGMA.input.txt
awk '{print $12,$2,$3,$9,$11}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_birth_weight/Fetal_BW_European_meta.NG2019.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_birth_weight/Fetal_BW_European_meta.NG2019_MAGMA.input.txt
awk '{print $3,$1,$2,$9,$10}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_head_circumference/EGG_HC_DISCOVERY.v2.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_head_circumference/EGG_HC_DISCOVERY.v2_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_female_late/Pubertal_growth_PTF.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_female_late/Pubertal_growth_PTF_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_female_SDS/Pubertal_growth_PGF.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_female_SDS/Pubertal_growth_PGF_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_late_total/Pubertal_growth_PTF_PTM_combined.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_late_total/Pubertal_growth_PTF_PTM_combined_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_male_late/Pubertal_growth_PTM.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_male_late/Pubertal_growth_PTM_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_male_SDS/Pubertal_growth_PGM.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_male_SDS/Pubertal_growth_PGM_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_total/Pubertal_growth_PGF_PGM_combined.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_growth_total/Pubertal_growth_PGF_PGM_combined_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_female_SDS/Pubertal_growth_10F.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_female_SDS/Pubertal_growth_10F_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_male_SDS/Pubertal_growth_12M.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_male_SDS/Pubertal_growth_12M_MAGMA.input.txt
awk '{print $1,$2,$3,$8,$9}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_total/Pubertal_growth_10F_12M_combined.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/EGG_pubertal_height_total/Pubertal_growth_10F_12M_combined_MAGMA.input.txt

# GEFOS cohorts
awk '{print $3,$1,$2,$9,$14}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_tblhBMD/bivariate_final.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_tblhBMD/bivariate_final_MAGMA.input.txt
awk '{print $3,$1,$2,$12,$18}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_faBMD/fa2stu.MAF0_.005.pos_.out_ > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_faBMD/fa2stu.MAF0_.005.pos_.out_MAGMA.input.txt
awk '{print $3,$1,$2,$12,$18}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_fnBMD/fn2stu.MAF0_.005.pos_.out_ > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_fnBMD/fn2stu.MAF0_.005.pos_.out_MAGMA.input.txt
awk '{print $3,$1,$2,$12,$18}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_lsBMD/ls2stu.MAF0_.005.pos_.out_ > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GEFOS_lsBMD/ls2stu.MAF0_.005.pos_.out_MAGMA.input.txt

# GIANT cohorts - need chromsomes added using biomart. Also rounded N to whole numbers and added N-case and N-controls for extreme height.
awk '{print $1,$10,$11,$7,$12}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_extreme_height/GIANT_EXTREME_HEIGHT_Stage1_Berndt2013_publicrelease_HapMapCeuFreq_wCHR.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_extreme_height/GIANT_EXTREME_HEIGHT_Stage1_Berndt2013_publicrelease_HapMapCeuFreq_MAGMA.input.txt
awk '{print $1,$9,$10,$7,$8}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_male/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N_wCHR.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_male/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_MEN_N_MAGMA.input.txt
awk '{print $1,$9,$10,$7,$8}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_female/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N_wCHR.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_female/GIANT_Randall2013PlosGenet_stage1_publicrelease_HapMapCeuFreq_HEIGHT_WOMEN_N_MAGMA.input.txt
awk '{print $1,$9,$10,$7,$8}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_total/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq_wCHR.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/GIANT_height_total/GIANT_HEIGHT_Wood_et_al_2014_publicrelease_HapMapCeuFreq_MAGMA.input.txt

# UKBB cohorts. OA stats as they dont have N or RSID numbers. Very challenging to use.
awk '{print $2,$3,$4,$11,$14}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/Biobank2-British-Bmd-As-C-Gwas-SumStats.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/Biobank2-British-Bmd-As-C-Gwas-SumStats_MAGMA.input.txt
awk '{print $2,$3,$4,$14,$17}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_fracture/Biobank2-British-FracA-As-C-Gwas-SumStats.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_fracture/Biobank2-British-FracA-As-C-Gwas-SumStats_MAGMA.input.txt
# awk '{print $12,$2,$3,$9,$11}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_OA_all/Tachmazidou_30664745_ALLOA.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_OA_all/Tachmazidou_30664745_ALLOA_MAGMA.input.txt
# awk '{print $12,$2,$3,$9,$11}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_OA_hospital/Tachmazidou_30664745_HIPKNEEOA.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_OA_hospital/Tachmazidou_30664745_HIPKNEEOA_MAGMA.input.txt
awk '{print $3,$1,$2,$9,$10}' /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_height/Meta-analysis_Locke_et_al+UKBiobank_2018.txt > /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_height/Meta-analysis_Locke_et_al+UKBiobank_2018_MAGMA.input.txt






/share/ScratchGeneral/scoyou/local/bin/MAGMA/magma --annotate window=1 --snp-loc /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/Biobank2-British-Bmd-As-C-Gwas-SumStats_MAGMA.input.txt --gene-loc /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/MAGMA_files/NCBI37/NCBI37.3.gene.loc --out /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/UKBB.eBMD_NCBI37.3
/share/ScratchGeneral/scoyou/local/bin/MAGMA/magma --bfile /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/MAGMA_files/g1000_eur/g1000_eur --gene-annot /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/UKBB.eBMD_NCBI37.3.genes.annot --gene-model snp-wise=mean --gene-settings snp-min-maf=0.000001 --seed 1 --pval /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/Biobank2-British-Bmd-As-C-Gwas-SumStats_MAGMA.input.txt use=RSID,P duplicate=first ncol=N --out /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_mean
/share/ScratchGeneral/scoyou/local/bin/MAGMA/magma --bfile /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/MAGMA_files/g1000_eur/g1000_eur --gene-annot /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/UKBB.eBMD_NCBI37.3.genes.annot --gene-model snp-wise=multi --gene-settings snp-min-maf=0.000001 --seed 1 --pval /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/Biobank2-British-Bmd-As-C-Gwas-SumStats_MAGMA.input.txt use=RSID,P duplicate=first ncol=N --out /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_multi
/share/ScratchGeneral/scoyou/local/bin/MAGMA/magma --gene-results /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_mean.genes.raw --set-annot /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/gene_sets/Dysplasia_ENTREZ.txt --out /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_mean
/share/ScratchGeneral/scoyou/local/bin/MAGMA/magma --gene-results /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_multi.genes.raw --set-annot /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/gene_sets/Dysplasia_ENTREZ.txt --out /share/ScratchGeneral/scoyou/projects/dysplasia_GWAS/skeletal_traits/UKBB_eBMD/skeletal_traits/UKBB_eBMD_multi

